frontend/udp/frontend.go

```diff
@@ -66,6 +68,20 @@ type Frontend struct {
 
 // NewFrontend allocates a new instance of a Frontend.
 func NewFrontend(logic frontend.TrackerLogic, cfg Config) *Frontend {
+	// Generate a private key if one isn't provided by the user.
+	if cfg.PrivateKey == "" {
+		log.Warning("no UDP private key provided: one will be generated for this instance")
+
+		buf := make([]byte, 2048)
+		_, err := rand.Read(buf)
+		if err != nil {
+			panic("system failed to provide sufficient randomness: " + err.Error())
+		}
+		cfg.PrivateKey = string(buf)
+
+		log.Info("the following UDP private key was generated: %s", cfg.PrivateKey)
```

Comment:
So this has blocked this PR for a bit. Should we enforce that this is a string rather than []byte? There isn't a crazy requirement around cryptographic security for this value.